lib/common/streams/batchoperation.js

```diff
@@ -112,25 +114,27 @@ BatchOperation.prototype.IsWorkloadHeavy = function() {
   if(this.enableReuseSocket && !this.callInOrder) {
     sharedRequest = 2;
   }
-  return this._activeOperation >= sharedRequest * this.concurrency ||
-    this._isLowMemory() ||
-    (this._activeOperation >= this.concurrency && this._getApproximateMemoryUsage() > 0.5 * SystemTotalMemory);
+  return this._activeOperation >= sharedRequest * this.concurrency ||this._isLowMemory();
 };
 
 /**
-* get the approximate memory usage for batch operation
+* Get the approximate memory usage for batch operation.
 */
 BatchOperation.prototype._getApproximateMemoryUsage = function() {
-  var currentUsage = process.memoryUsage().rss;
+  var currentUsage = azureutil.isBrowser() ? 0 : process.memoryUsage().rss; // Currently, we cannot get memory usage in browsers
```

Comment:
Actually, it's a little difficult to estimation the memory usage without browsers' API support. Different web pages may use different size of memories. Anyway, to align with the truth, will add a constant to estimate current's memory usage. But the value of the constant maybe small, such as several MBs.